src/main/java/org/tugraz/sysds/runtime/controlprogram/caching/TensorObject.java

```diff
@@ -38,7 +38,7 @@
 
 import java.io.IOException;
 
-public class TensorObject extends CacheableData<TensorBlock>
+public class TensorObject extends CacheableData<HomogTensor>
```

Comment:
No, the tensor object should be using TensorBlock (i.e., the abstract class above BasisTensor and DataTensor) - otherwise this entire class hierarchy has no use.